src/query/graphite/native/aggregation_functions.go

```diff
@@ -135,6 +135,39 @@ func divideSeries(ctx *common.Context, dividendSeriesList, divisorSeriesList sin
 	return r, nil
 }
 
+// aggregate takes a list of series and returns a new series containing the
+// value aggregated across the series at each datapoint using the specified function.
+func aggregate(ctx *common.Context, series singlePathSpec, fname string) (ts.SeriesList, error) {
+	switch fname {
+	case Empty, Sum, SumSeries, Total:
+		return sumSeries(ctx, multiplePathSpecs(series))
+	case Min, MinSeries:
+		return minSeries(ctx, multiplePathSpecs(series))
+	case Max, MaxSeries:
+		return maxSeries(ctx, multiplePathSpecs(series))
+	case Avg, Average, AverageSeries:
+		return averageSeries(ctx, multiplePathSpecs(series))
+	case Multiply, MultiplySeries:
+		return multiplySeries(ctx, multiplePathSpecs(series))
+	case Diff, DiffSeries:
+		return diffSeries(ctx, multiplePathSpecs(series))
+	case Count, CountSeries:
+		return countSeries(ctx, multiplePathSpecs(series))
+	case Range, RangeOf, RangeOfSeries:
+		return rangeOfSeries(ctx, series)
+	case Last, KeepLastValue:
+		return keepLastValue(ctx, series, -1) // using default value, same as graphite-web python implementation.
+	case Stdev, Stddev:
+		// default value for points in python implementation not clear, using an arbitrary 5 data points value should works fine.
+		// default value for null value ratio is same as graphite-web python implementation.
+		return stdev(ctx, series, 5, 0.1)
```

Comment:
So our stddev function looks like it does something different than normal graphite. Normal graphite looks like it takes the standard deviation across all time series, rather than the time series per series. 

Perhaps we should do that for the aggregate function? 
https://github.com/graphite-project/graphite-web/blob/3ad279df5cb90b211953e39161df416e54a84948/webapp/graphite/functions/safe.py#L86-L96

In fact all series that are returned from "aggregate" in graphite-web seem to only return a single value, so should make sure that all these return just a single series.

Can see this in effect here:
https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/render/functions.py#L195-L197
```py
  series = TimeSeries(name, start, end, step, values, xFilesFactor=xFilesFactor, tags=tags)

  return [series]
```

Reply:
Thanks for catching this. will take another look at the stddev func and another pass of all these returns.

Reply:
I updated this to use the same StdDev logic as the `graphite-web` source code.